packages/openapi3/src/openapi.ts

```diff
@@ -1097,13 +1097,49 @@ function createOAPIEmitter(program: Program, options: ResolvedOpenAPI3EmitterOpt
     );
   }
 
+  function checkAdditionalProperties(program: Program, model: Model): boolean {
+    const baseModel = model.baseModel;
+    if (!baseModel || baseModel.name !== "Record") {
+      return false;
+    }
+    const templateArg = baseModel.templateMapper?.args.at(0);
+    if (!templateArg) {
+      return false;
+    }
+    switch (templateArg.kind) {
+      case "Intrinsic":
+        return templateArg.name === "unknown";
+      case "Scalar":
+        let isAssignable = true;
+        for (const [_, prop] of model.properties) {
+          const [isTypeAssignable, diagnostics] = program.checker.isTypeAssignableTo(
+            prop.type,
+            templateArg,
+            prop
+          );
```

Comment:
@nguerrera @bterlson the problem might also lie here. Do you see any obvious issue?